File: 2016/day3/pa.lua
local valid = 0

for line in io.lines("./input.txt") do
	local t = {}

	for n in line:gmatch("%d+") do
		t[#t + 1] = tonumber(n)
	end

	if #t == 3 then
		table.sort(t)
		local a, b, c = t[1], t[2], t[3]

		if a + b > c then
			valid = valid + 1
		end
	end
end

print(valid)

File: 2016-py/day3/pa.py
def main():
    valid = 0

    with open("input.txt", "r") as fin:
        for line in fin:
            sides = []

            for n in line.split():
                sides.append(int(n))

            if len(sides) == 3:
                sides.sort()
                a, b, c = sides

                if a + b > c:
                    valid += 1

    print(valid)


if __name__ == "__main__":
    main()

